### PR TITLE
fix(dev): add hot reloading

### PR DESCRIPTION
**Pull request by psavery**
This adds a `--hot-reload` option where, if set, controller/state callback functions will be automatically reloaded for every function call. This excludes functions that are located in site-packages directories (which are usually libraries that the user is not currently developing).

There is also a `@hot_reload` decorator that may be added to functions as well, which will cause the function to be reloaded every time.

This work is largely based off of https://github.com/julvo/reloading, with some major modifications, including adding support for methods. His license is included within the file.

### CODE REVIEW

File: trame_server/controller.py [hunk 1]
```diff
@@ -29,10 +30,21 @@ class Controller:
...
-    def __init__(self, trigger_decorator, trigger_name):
+    def __init__(self, server):
+        super().__setattr__("_server", server)
         super().__setattr__("_func_dict", {})
-        self.trigger = trigger_decorator
-        self.trigger_name = trigger_name
+
+    @property
+    def trigger(self):
+        return self._server.trigger
```
Comment (jourdain) [inline trame_server/controller.py:39]: Those are functions. Are they going to work when exposed as properties?

File: trame_server/controller.py [hunk 2]
```diff
@@ -29,10 +30,21 @@ class Controller:
...
+
+    @property
+    def trigger_name(self):
+        return self._server.trigger_name
```
Comment (jourdain) [inline trame_server/controller.py:43]: Those are functions. Are they going to work when exposed as properties?

File: trame_server/controller.py (hunk 1 above)
Comment (psavery) [reply trame_server/controller.py:39]: Yes they do.

File: trame_server/controller.py (hunk 2 above)
Comment (psavery) [reply trame_server/controller.py:43]: Yes they do.

**Pull request merged**